tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -715,6 +716,13 @@ private void OnRoutingEndpointMatched(object arg)
                 var request = httpContext.Request.DuckCast<HttpRequestStruct>();
                 RouteValueDictionary routeValues = request.RouteValues;
 
+                var security = Security.Instance;
+                var shouldSecure = security.Settings.Enabled;
+                if (shouldSecure)
+                {
+                    security.InstrumentationGateway.RaisePathParamsAvailable(httpContext, span, HttpRequestUtils.ConvertRouteValueDictionary(routeValues));
```

Comment:
If you're using MVC/Web API on .NET Core 3.0+ you're calling `RaisePathParamsAvailable` twice - once here, and once in `OnMvcBeforeAction`. I think you want to check `IsUsingEndpointRouting` to see whether to raise this again (unless it's a noop the second time, and doesn't hit perf)?

Reply:
Yes indeed it's raised twice! and we want the `OnMvcBeforeAction ` to raise here, I couldnt find a way to detect,  for the first time, at routing endpoint matched, to not do anything if it's mvc/web api, that means, if later we're going in mvc before action for this request... The difference is, the second time, the route values we get are the real ones, without {controller} and {action} keys.. And I didnt want to filter them out myself, cause we _might_ have route params that are actually route param and could be named controller or action (yes it s a bit far fetched..), wanted to let mvc do the job and give us the real params... so the second time will erase the results of the first time in case of mvc and that's wanted. Because I couldnt find a way not to execute the first time for these ...  not sure if I'm clear 🤯

Reply:
Ah, ok, yeah that makes sense 🙂 Technically I'm not sure if there's a ways to detect that. You can probably detect whether it _might_ run, but e.g. if there's authorization middleware, then the final MVC might not run at all, so your approach seems good to me 👍